routes/produce.js

```diff
@@ -51,12 +53,194 @@ router.get(
         message: 'You are not authorised to view this resource.',
         title: 'Error',
         user: req.user,
+        filter_data: '',
         page_name: 'error',
       });
     }
   }
 );
 
+//Filter for Produce Page
+router.get(
+  '/filter/:range',
+  require('connect-ensure-login').ensureLoggedIn({ redirectTo: '/app/auth/login' }),
+  function (req, res, next) {
+    if (req.user.role === ROLES.Admin && ROLES.Superuser) {
```

Comment:
This should be admin or superuser i.e. `(req.user.role === ROLES.Admin || ROLES.Superuser)`